tools/clang/lib/Format/TokenAnnotator.cpp

```diff
@@ -1667,7 +1667,7 @@ unsigned TokenAnnotator::splitPenalty(const AnnotatedLine &Line,
   if (Left.is(tok::coloncolon) ||
       (Right.is(tok::period) && Style.Language == FormatStyle::LK_Proto))
     return 500;
-  if (Left.isOneOf(tok::kw_class, tok::kw_struct))
+  if (Left.isOneOf(tok::kw_class, tok::kw_struct, tok::kw_union))
```

Comment:
This change isn't required for C++ to support unions, so I don't understand why it is required for HLSL.

Reply:
Done. New pull request has been created. Here is the link :https://github.com/microsoft/DirectXShaderCompiler/pull/4132